pointcloud/pointcloud_file.go

```diff
@@ -494,6 +494,25 @@ func ReadPCDToKDTree(inRaw io.Reader) (*KDTree, error) {
 	return kd, nil
 }
 
+// ReadPCDToBasicOctree reads a PCD file into a basic octree.
+func ReadPCDToBasicOctree(inRaw io.Reader) (*BasicOctree, error) {
+	pc, err := ReadPCD(inRaw)
+	if err != nil {
+		return nil, err
+	}
+
+	basicOctPC, err := ConvertPointCloudToBasicOctree(pc)
```

Comment:
In slack, you discussed avoiding creating a temporary PointCloud: https://viaminc.slack.com/archives/C036N5GJFLM/p1673639600522369. Did that not work out?

Reply:
That was a misunderstanding on my part. I forgot the order in which I gave the options and thought you gave your approval for the octree via pointcloud version. 

The only issue with the max/min values is that the helper functions for pointclouds includes various versions based on the data type defined in the header, specifically `readPCDAscii` and `readPCDBinary`. These would either have to be edited to also return a list of values (representing max min) or have new functions created that repeat the same process, but with without the final Set() prodecure (see line [here](https://github.com/viamrobotics/rdk/blob/717b44adb0c60ebf51e541340ba60fce52bb472d/pointcloud/pointcloud_file.go#L563)). This set procedure only takes O(1) for non-octree pointclouds so its actually more efficient processing wise to create the pointcloud as we would have to re-parse the pcd file, only iterate through the temp pointcloud. 

Therefore the biggest con to the pointcloud method is just memory, not time or processing power. However, I am happy to create/edit the helper functions to return the max and min values if memory is more important than complexity and simplicity.

Reply:
Here is an example of this: [PR #1774](https://github.com/viamrobotics/rdk/pull/1774/files#diff-e01dea69a2aaa9209cb5deeb1c4569f99ef8c51262c69799a0cec3a79d7434eb)

Reply:
We could also consider combining `getPCDMetaDataBinary` and `readPCDBinary` using a boolean input and also returning the `[]float64`

Reply:
I think we should avoid the memory use of creating a PointCloud, particularly since garbage collection will need to clean this up. I'm not so familiar with how we might share code with the pointcloud implementation, but I hope it will be easier now that we're moving Octree into the pointcloud package.